packages/base/src/endpoint.ts

```diff
@@ -1,5 +1,5 @@
 export interface Endpoint {
-  channel: { name: string; version: string }
+  channel: string | { name: string; version: string }
   identity: string
```

Comment:
If the channel value is a string then it's a custom channel. If it's an object with a version and name then it's a builtin channel

Reply:
Why not just add a custom property or something?

Reply:
It's for simplicity. You're rarely ever going to do custom mapping for a builtin channel (can't even think of a scenario where that would make sense), so for 99.9% of use cases it's simpler for it to just be a string